### PR TITLE
If no group DN is configured, still look for policies on local users and

### PR DESCRIPTION
**Pull request by jefferai**
return a warning, rather than just trying to do an LDAP search on an
empty string.

### CODE REVIEW

File: builtin/credential/ldap/backend.go [hunk 1]
```diff
@@ -183,10 +191,11 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 	}
 
 	if len(policies) == 0 {
-		return nil, logical.ErrorResponse("user is not member of any authorized group"), nil
+		resp.Data["error"] = "user is not a member of any authorized group"
```
Comment (vishalnayak) [inline builtin/credential/ldap/backend.go]: The sting "no group DN..." will be replicated in both `warnings` field and in resp.Data["error"] field. If you left it knowingly, then its fine. I only wanted to bring your attention on this.

Initially we were error-ing. Now that we are not, policies and error are nil here.
But, we are doing `if len(policies) == 0 { return resp, err }` in the calling function.
I feel it should be `if policies == nil { return resp, nil }`.

Comment (vishalnayak) [inline builtin/credential/ldap/backend.go]: Okay. You have told me already that you added it both in `warnings` and `errors` intentionally. I had overlooked it.

**Comment by sigmunau** [comment]
This is actually a nice feature. This allows us to create a specific ldap mount where groups are only managed through vault, not risking that someone managed to convince the ldap administrator to create a group that happens to have the name of a group configured in vault.

**Comment by jefferai** [comment]
Glad it's useful for you!

**Pull request merged**